lib/pure/httpclient.nim

```diff
@@ -980,7 +980,7 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url, httpMethod: string,
   var data: seq[string]
   if multipart != nil and multipart.content.len > 0:
     data = await client.format(multipart)
-  else:
+  elif not (httpMethod == "GET" and body.len == 0):
```

Comment:
I think it might be better to be more explicit here:
```suggestion
  elif httpMethod in ["POST", "PATCH", "PUT"] or body.len != 0:
```

I'm not sure if that's all the HTTP methods that should send content-length even when it's empty, but certainly it's not just "GET" that should avoid this logic.

Reply:
Your suggestion works for me. I would argue its more correct. Its a bigger change. I was hesitant to propose it at first.